mnelab/viz.py

```diff
@@ -19,30 +21,110 @@ def _get_rows_cols(n):
     return rows, cols
 
 
-def plot_erds(data, freqs, n_cycles, baseline, times=(None, None)):
-    tfr = tfr_multitaper(data, freqs, n_cycles, average=False, return_itc=False)
+def calc_tfr_and_masks(epochs, freqs, baseline, times, alpha=None):
+    """
+    Calculate AverageTFR and significance masks for given epochs.
+
+    Adapted from https://mne.tools/dev/auto_examples/time_frequency/time_frequency_erds.html
+
+    Parameters
+    ----------
+    epochs : mne.epochs.Epochs
+        Epochs extracted from a Raw instance.
+    freqs : np.ndarray
+        The frequencies in Hz.
+    baseline : array_like, shape (2,)
+        The time interval to apply rescaling / baseline correction.
+    times : array_like, shape (2,)
+        Start and end of crop time interval.
+    alpha : float, optional
+        If specified, calculate significance maps with threshold `alpha`, by default `None`.
+
+    Returns
+    -------
+    dict[str, tuple[mne.time_frequency.tfr.EpochsTFR, dict[str, np.ndarray | None]]]
+        A dictionary where keys are event IDs and values are tuples (`tfr_ev`, `masks`).
+        `tfr_ev` is the EpochsTFR object for the respective event. `masks` is again a
+        dictionary, where keys are channel names and values are significance masks.
+        Significance masks are `None` if `alpha` was not specified.
+    """
+    tfr = tfr_multitaper(epochs, freqs, freqs, average=False, return_itc=False)
     tfr.apply_baseline(baseline, mode="percent")
     tfr.crop(*times)
 
+    pcluster_kwargs = dict(
+        n_permutations=100,
+        step_down_p=0.05,
+        seed=1,
+        buffer_size=None,
+        out_type='mask',
+    )
```

Comment:
Except for alpha, I left the pcluster parameter hardcoded as in the [MNE-Python example](https://mne.tools/stable/auto_examples/time_frequency/time_frequency_erds.html). Do you want to expose some of them?

Reply:
Hm, I don't think that's necessary, at least for now. Did you see that there is an `n_jobs` parameter? Should we use it and would that work in combination with starting in a separate process?

Reply:
I did, but didn't think further about it since joblib isn't a (direct) dependency of mnelab at the moment... but will try that!

Reply:
Apparently not: `UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1`. Do you want to add that as an option nonetheless?

Reply:
Nah, if it doesn't work we don't need it. I'd rather have a dialog which can cancel the computation.